src/client/caching/structures/Message.ts

```diff
@@ -192,6 +194,28 @@ export class Message extends Structure {
 		return this.editedTimestamp ? new Date(this.editedTimestamp) : null;
 	}
 
+	/**
+	 * Edits the message.
+	 * @param content The {@link MessageBuilder builder} to send.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#edit-message
+	 */
+	public async edit(content: MessageBuilder): Promise<Message> {
+		const data = await this.client.api.patch(Routes.channelMessage(this.channel.id, this.id), content) as APIMessageData;
+		return this._patch(data) as Message;
+	}
+
+	/**
+	 * Deletes the message.
+	 * @param reason The reason to write in the audit logs.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#delete-message
+	 */
+	public async delete(reason: string): Promise<this> {
+		await this.client.api.delete(Routes.channelMessage(this.channel.id, this.id), { reason });
```

Comment:
Does reason work like this? If yes, why isn't ban using this?

Reply:
Replied in the respective question.
And yes, it works like that, `@klasa/rest` takes that as an option and sets the header when it exists, alongside a few more options (query, data...).

Reply:
We should probably mark the Message as deleted here, to avoid any possible race conditions.